Clarify optional fmi3EvaluateDiscreteStates function call (#2178)
modelica/fmi-standard: docs/2_3_common_states.adoc

Not allowed in Co-Simulation and Scheduled Execution.

Function <<fmi3EvaluateDiscreteStates>>::
This function may only be called if <<providesEvaluateDiscreteStates>> is `true`.
It explicitly evaluates latexmath:[\mathbf{f}_{\mathit{disc}}], computing the current discrete states.
Calling this function is optional and the FMU must not rely on it being called.

> If it is omitted, latexmath:[\mathbf{f}_{\mathit{disc}}] is evaluated implicitly by <<fmi3UpdateDiscreteStates>> at the end of the super-dense time instant.

Reviewer: ```suggestion
If this explicit call is omitted in this super dense time instant, latexmath:[\mathbf{f}_{\mathit{disc}}] is evaluated implicitly by <<fmi3UpdateDiscreteStates>>.
```

> If this explicit call is omitted in this super dense time instant, latexmath:[\mathbf{f}_{\mathit{disc}}] is evaluated implicitly by <<fmi3UpdateDiscreteStates>>.

Reviewer: ```suggestion
It evaluates latexmath:[\mathbf{f}_{\mathit{disc}}], computing the current discrete states.
Calling this function is optional and the FMU must not rely on it being called.
If this call is omitted in a super dense time instant, latexmath:[\mathbf{f}_{\mathit{disc}}] is evaluated implicitly by <<fmi3UpdateDiscreteStates>>.
```
